src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,177 @@
+        for(Integer j = left.intValue(); j<=right.intValue(); j++){
+            range.add(j);
+        }
+        return range;
+
+    }
+
+    public static Range<Float> of(Float a, Float b) {
+        Range<Float> range = new Range<>();
+        range.clear();
+        left = a;
+        right = b;
+        for(Float j = left.floatValue(); j<=right.floatValue(); j=j+0.1f){
+            range.add(j);
+        }
+        return range;
+
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> characterCharacterFunction) {
+        Range<Character> range  = new Range<>();
+        range.clear();
+        for(Character j = a; j<=d; j++){
+            range.add(j);
+        }
+        return range;
+    }
 
     public int size() {
-        return 0;
+        return modCount;
     }
 
     public boolean isEmpty() {
-        return false;
+        return this.isEmpty;
     }
 
     public boolean contains(Object o) {
-        return false;
+        return java.util.Arrays.asList(elementData).indexOf(o) >= 0;
     }
 
     public Iterator<T> iterator() {
-        return null;
+        Iterator iterator = new Iterator() {
+            int cursor = 0;
+            int lastRet = -1;
+            @Override
+            public boolean hasNext() {
+                return cursor != modCount;
+            }
+
+            @Override
+            public Object next() {
+                int i = cursor;
+                if (i >= modCount)
+                    throw new NoSuchElementException();
+                if (i >= elementData.length)
+                    throw new ConcurrentModificationException();
+                cursor = i + 1;
+                return (Object) elementData[lastRet = i];
+            }
+        };
+        return iterator;
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        return elementData;
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+        T1[] array = (T1[]) new Object[a.length];
+        int i = 0;
+        for(T1 element: a){
+            array[i] = element;
+            i++;
+        }
+        return array;
     }
 
+
     public boolean add(T t) {
-        return false;
+
+        if(java.util.Arrays.asList(elementData).indexOf(t) >= 0){
+            return false;
+        }
+
+        elementData[modCount] = t;
+        modCount++;
+        if(this.isEmpty){
+            this.isEmpty = false;
+        }
+        if(modCount == 1){
+            this.isEmpty = true;
+        }
+        return true;
     }
 
     public boolean remove(Object o) {
-        return false;
+
+        int index = java.util.Arrays.asList(elementData).indexOf(o);
+        for (int i = index; i<modCount; i++){
+            elementData[i] = elementData[i+1];
+        }
+        modCount--;
+        return true;
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        boolean flag = true;
+        for (Object element: c) {
```

Comment:
If you don't find an element, you can stop, and don't check more by using while loop or just break this loop.